Lib/fontTools/ttLib/tables/G__l_a_t.py

```diff
@@ -139,11 +139,11 @@ def compile(self, ttFont):
         return data
 
     def compileAttributes12(self, attrs, fmt):
-        data = []
+        data = b""
         for e in grUtils.entries(attrs):
-            data.extend(sstruct.pack(fmt, {'attNum' : e[0], 'num' : e[1]}))
-            data.extend(struct.pack(('>%dh' % len(e[2])), *e[2]))
-        return "".join(data)
+            data += sstruct.pack(fmt, {'attNum' : e[0], 'num' : e[1]})
+            data += struct.pack(('>%dh' % len(e[2])), *e[2])
```

Comment:
appending and then joining once at the end of the loop is generally more efficient.
there's a `bytesjoin` function in fontTools.misc.py23 which you could use here

Reply:
Yeh. I did that as a lazy man's pep8. But I notice that line length isn't a strong value. Anyway I have tidied this up. I don't think a malloc + memcpy is that expensive over managing an array.